### ISSUE TITLE
DNN, Keras model included Flatten() problem.

### ISSUE THREAD

**Issue by sangjoonhong** [opencv/opencv#15141]
- OpenCV => 4.1.0
- Operating System / Platform => Ubuntu 16.04

Hello, 
I have a problem using dnn module.

I made binary classification model using Keras.
I used pre-trained mobilenet v2 (imagenet)

And this is part of my code:

```
conv_base = MobileNetV2(weights='imagenet', include_top=False, input_shape=(96, 96, 3))
model = models.Sequential()
model.add(conv_base) # mobilenetv2
model.add(layers.Reshape([3 * 3 * 1280])) # Flatten() -> Reshape()
model.add(layers.Dense(256, activation='relu'))
model.add(layers.Dropout(0.5))
model.add(layers.Dense(1, activation='sigmoid'))
```

Of course, Error is occur
OpenCV(4.1.0) C:\projects\opencv-python\opencv\modules\dnn\src\dnn.cpp:524: error: (-2:Unspecified error) Can't create layer "strided_slice" of type "StridedSlice" in function 'cv::dnn::dnn4_v20190122::LayerData::getLayerInstance'

**I found many questions about same problem. 
But, change the 'Flatten()' to 'Reshape()' is not working solution.
I changed but, same error is still occur.**

I saw reshape is only working in Keras version 2.0.8
But I can't using that version because using pretrained mobilenetv2.
It is not exist in Keras version 2.0.8

Is there any other solution about this problem?
Thank you.

**Comment by dkurt** [comment]
Pleas follow issues template and make it easier to reproduce.

**Comment by sangjoonhong** [comment]
https://answers.opencv.org/question/183682/opencv-dnn-import-dropout-layer-error-after-finetuning-keras-vgg16/
I found similar question and answer.
But I don't know how apply to my problem.
Nodes' components are little different to that article

**Comment by dkurt** [comment]
@sangjoonhong, Please provide at least a `.pb` model.

**Comment by sangjoonhong** [comment]
https://drive.google.com/open?id=1JE_P4OmCOF4vhQvstQMvJIxSPFna5bFG

This is my .pb model, using Flatten()

**Comment by sangjoonhong** [comment]
Is there any updates? :)

**Comment by dkurt** [comment]
@sangjoonhong, Thanks for report! Please try the changes from https://github.com/opencv/opencv/pull/15391 (with Reshape model for now)

test code:
```python
import tensorflow as tf
import cv2 as cv
import numpy as np

model = 'normal_and_abnormal_mnet_v2_96_96_Reshape.pb'

np.random.seed(324)
inp = np.random.uniform(0.0, 1.0, [1, 96, 96, 3]).astype(np.float32)

with tf.gfile.FastGFile(model, 'rb') as f:
    graph_def = tf.GraphDef()
    graph_def.ParseFromString(f.read())

with tf.Session() as sess:
    sess.graph.as_default()
    tf.import_graph_def(graph_def, name='')

    tfOut = sess.run(sess.graph.get_tensor_by_name('dense_2/Sigmoid:0'),
                     feed_dict={'mobilenetv2_1.00_96_input:0': inp})

net = cv.dnn.readNet(model)
net.setInput(inp.transpose(0, 3, 1, 2))  # NHWC to NCHW
cvOut = net.forward()
print cvOut
print tfOut
print np.max(np.abs(cvOut - tfOut))
```


output:
```
[[ 0.64483404]]
[[ 0.6448338]]
2.38419e-07
```

**Comment by sangjoonhong** [comment]
@dkurt Thank you very much :)
